Theme JSON Docs: Clarify null, true, and false values for blockGap setting (#42447)
WordPress/gutenberg: docs/how-to-guides/themes/theme-json.md

### What is blockGap and how can I use it?

blockGap adjusts the vertical margin, or gap, between blocks.
It is also used for margins between inner blocks in columns, buttons, and social icons.
In the editor, the control for the blockGap is called Block spacing, located in the Dimensions panel.

The value you define for the blockGap style uses a CSS property, a preset, named `--wp--style--block-gap`.

> The default value is 2em.

Author: I've removed this line about the default value, as it will be different per theme, and the core `theme.json` shipped in the editor has a different default (currently `24px`) so specifying the particular value in the docs could cause confusion. Instead, at the bottom of this changeset I've added a line that the root `blockGap` value will be output to the CSS variable.

Reviewer: I think we need to make it even clearer what is supported by core "block editor" in 6.0 and what is supported in the active Gutenberg plugin.

Reviewer: it should not block this PR though -I think we need to reconsider what information should go into the theme developer handbook and what should be on this page and even split it into subpages.

Author: Thanks for taking a look @carolinan! Yes, it'd be great to explore options there in follow-ups, so that we have some neater areas for expanding on how different functionality works, particularly as the features keep growing / becoming more complex.

I'd be keen to do a follow-up PR at some point to expand on the current state of Layout support, but it'd probably need to be specific to "here's how it works in Gutenberg right now" so it'd be good to figure out how to neatly flag / present that in a way that's approachable for folks who regularly use these docs. Let me know if you have any suggestions and I'm happy to have a go at a follow-up PR with some extra context next week.

}
```

> The setting for `blockGap` is a boolean, and its default value is `null`. This allows an extra level of control over style output. The `settings.spacing.blockGap` setting in a `theme.json` file accepts the following values:

Reviewer 2: ```suggestion
The setting for `blockGap` is either a boolean or `null` value and is `null` by default. This allows an extra level of control over style output. The `settings.spacing.blockGap` setting in a `theme.json` file accepts the following values:
```

Nit: A small wording change that might make the type of accepted values for `blockGap` clearer again.

> - `true`: Opt in to displaying _Block spacing_ controls in the editor UI, and output `blockGap` styles.

Reviewer 2: ```suggestion
- `true`: Opt into displaying _Block spacing_ controls in the editor UI and output `blockGap` styles.
```

Nit: "in to" should generally be combined to "into" when it indicates movement or a change of some kind.

Author: Oh, good point. I found myself staring at `opt-in` `opt in` and `opt into` for far too long, thanks for giving an answer! 😄

Reviewer 2: I'm in that same situation most days. Before adding the suggestion I did have to go and ask the grammar gods 😁

> - `false`: Opt out of displaying _Block spacing_ controls in the editor UI, with `blockGap` styles stored in `theme.json` still being rendered. This allows themes to make use of `blockGap` values, without allowing users to make changes within the editor.

Reviewer 2: ```suggestion
- `false`: Opt out of displaying _Block spacing_ controls in the editor UI, with `blockGap` styles stored in `theme.json` still being rendered. This allows themes to use `blockGap` values without allowing users to make changes within the editor.
```

Nit: Reduce wordy phrase and remove unnecessary comma.

> - `null` (default): Opt out of displaying _Block spacing_ controls, _and_ prevent output of `blockGap` styles.

Reviewer 2: ```suggestion
- `null` (default): Opt out of displaying _Block spacing_ controls, _and_ prevent the output of `blockGap` styles.
```

Nit: Missing determiner from noun phrase.
